light_curves/code_src/panstarrs_functions.py

```diff
@@ -244,53 +24,106 @@ def panstarrs_get_lightcurves(sample_table, *, radius=1/3600):
+        dec_column="dec_deg", 
+        margin_threshold=10,
+        # Optimize partition size
+        drop_empty_siblings=True
+    )
+
+    #plan to cross match panstarrs object with my sample 
+    #only keep the best match
+    yang_ps = panstarrs_object.crossmatch(
+        sample_lsdb, 
+        radius_arcsec=radius, 
+        n_neighbors=1, 
+        suffixes=("", "")
+
+    )
+        
+    # plan to join that cross match with detections to get light-curves
+    yang_ps_lc = yang_ps.join(
+        panstarrs_detect,
+        left_on="objID",
+        right_on="objID",
+         output_catalog_name="yang_ps_lc",
+        suffixes = ["",""]
+    )
+    
+    # Create default local cluster
+    # here is where the actual work gets done
+    with Client():
+        #compute the cross match with object table
+        #and the join with the detections table
+        matched_df = yang_ps_lc.compute()
     
-        #fix the column names to include filter names
-        dtab = addfilter(Table.read(dresults, format='ascii'))
-
-        dtab.sort('obsTime')
-        dtab['psfFlux'][dtab['psfFlux'] == -999.0] = np.nan
-        dtab['psfFluxErr'][dtab['psfFluxErr'] == -999.0] = np.nan
-
-        #here is the light curve mixed from all 5 bands
-        t_panstarrs = dtab['obsTime']
-        flux_panstarrs = dtab['psfFlux']*1E3  # in mJy
-        err_panstarrs = dtab['psfFluxErr'] *1E3
-        filtername = dtab['filter']
         
-        #put this single object light curves into a pandas multiindex dataframe
-        dfsingle = pd.DataFrame(dict(flux=flux_panstarrs, err=err_panstarrs, time=t_panstarrs, objectid=objectid, band=filtername, label=lab)).set_index(["objectid","label", "band", "time"])
-        #then concatenate each individual df together
-        df_lc.append(dfsingle)
-            
-    return(df_lc)
+    #cleanup the filter names to the expected letters
+    filter_id_to_name = {
+    1: 'Pan-STARRS g',
+    2: 'Pan-STARRS r',
+    3: 'Pan-STARRS i',
+    4: 'Pan-STARRS z',
+    5: 'Pan-STARRS y'
+    }
+    if len(matched_df["filterID"]) > 0:
+        get_name_from_filter_id = np.vectorize(filter_id_to_name.get)
+        filtername = get_name_from_filter_id(matched_df["filterID"])
+    else:
+        # Handle the case where the array is empty
+        filtername = []
+    
+    # setup to build dataframe 
+    t_panstarrs = matched_df["obsTime"]
+    flux_panstarrs = matched_df['psfFlux']*1E3  # in mJy
+    err_panstarrs = matched_df['psfFluxErr'] *1E3
+    lab = matched_df['label']
+    objectid = matched_df['objectid']
+
+    #make the dataframe of light curves
+    df_lc = pd.DataFrame(
+        dict(flux=pd.to_numeric(flux_panstarrs, errors='coerce').astype(np.float64), 
+             err=pd.to_numeric(err_panstarrs, errors='coerce').astype(np.float64), 
+             time=pd.to_numeric(t_panstarrs, errors='coerce').astype(np.float64), 
+             objectid=pd.to_numeric(objectid, errors='coerce').astype(np.int64), 
```

Comment:
Oh interesting. I wonder if it's fundamentally a result of choices made during the catalog import process rather than LSDB itself. I'll try to look at the data types more closely before I import another catalog to try to avoid these hurdles with the ones we put out.

Reply:
This difference between pyarrow/parquet vs pandas/numpy dtypes certainly came up before, and for end users I see all the reasons to provide the outputs in the latter, even if under the good it keeps mapping to what parquet likes.

Reply:
Good to know, thanks!